modules/cache/cache.go

```diff
@@ -35,6 +35,18 @@ func NewContext() error {
 		if conn, err = newCache(setting.CacheService.Cache); err != nil {
 			return err
 		}
+		const testKey = "__gitea_cache_test"
+		const testVal = "test-value"
+		if err = conn.Put(testKey, testVal, 10); err != nil {
+			return err
+		}
+		val := conn.Get(testKey)
+		if valStr, ok := val.(string); !ok || valStr != testVal {
```

Comment:
Hmm... I'm not sure that all caches promise this.

Imagine a shared cache sitting behind a cluster of Giteas.

Our new Gitea starts up and puts this in to the cache. 

The cache is already full and this is not a commonly seen entry.
This cache has a hidden unreported queue where it keeps track of keys and only evicts if a key has been seen twice therefore this put doesn't cause an eviction. So it's not cached. 

So the Get fails here even though the cache is working correctly.

Similarly there could be a race between a busy cache evicting the item and it being read back.

Potentially better to just leave this at the put or ensuring that the get doesn't panic.

Reply:
I will change the error to a warning.